hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+            filtered_res_list.append(resource_list_item)
+
+        return filtered_res_list
+
+
+class ResourceCreate(generics.CreateAPIView):
+    """
+    Create a new resource
+    REST URL: hsapi/resource
+    HTTP method: POST
+    HTTP status code (on success): 201
+    Request data payload parameters:
+    :type   resource_type: str
+    :type   title: str
+    :type   metadata: json
+    :type   edit_users: list
+    :type   edit_groups: list
+    :type   view_users: list
+    :type   view_groups: list
+    :param  resource_type: resource type name
+    :param  title: (optional) title of the resource
+    :param  metadata: (optional) dublin core and any extended metadata supported by resource_type
+    :param  edit_users: (optional) list of user ids that should have edit permission for the resource
+    :param  edit_groups: (optional) list of group ids that should have edit permission for the resource
+    :param  view_users: (optional) list of user ids that should have view permission for the resource
+    :param  view_groups: (optional) list of group ids that should have view permission for the resource
+    :return: id of the resource created
+    :rtype: json string of the format: {'status':'CREATED', 'status_code': 201, 'message':'Resource created',
+                                        'data':{'resource-id':id, 'resource_type': resource type}}
+    :raises:
+    &nbsp;&nbsp;&nbsp;NotAuthenticated: return json format: {'status': UNAUTHORIZED, 'status_code':401, 'message': 'Authentication
+                                            credentials were not provided.'}
+    &nbsp;&nbsp;&nbsp;HTTP status code:401
+    &nbsp;&nbsp;&nbsp;ValidationError: return json format: {'status': BAD REQUEST, 'status_code':400', 'message': {parameter-1':
+                                            'error message-1', 'parameter-2': 'error message-2', .. }
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    #parser_classes = (FileUploadParser, )
+    # @property
+    # def allowed_methods(self):
+    #     return ['POST']
+
+    def get_serializer_class(self):
+        return serializers.ResourceCreateRequestValidator
+
+    # Override the create() method from the CreateAPIView class
+    def create(self, request, *args, **kwargs):
+        rest_api_response = utils.HSRestAPIResponse()
+
+        if not request.user.is_authenticated():
+            raise NotAuthenticated()
+
+        resource_create_request_validator = serializers.ResourceCreateRequestValidator(data=request.data)
+        if not resource_create_request_validator.is_valid():
+            raise ValidationError(detail=resource_create_request_validator.errors)
+
+        resource_type = request.DATA['resource_type']
+        res_title = request.DATA.get('title', None)
+        if not res_title:
+            res_title = 'Untitled resource'
+        resource_files = request.FILES.values()
```

Comment:
I would like to suggest that this initial version of the API not include FILE upload with resource creation. We separately have an API to add individual files to resources once created, and it would be simpler to have one way to upload files instead of two.

This also helps to mitigate the fact that we can't currently support enormous multi-gigabyte request sizes, so if API users had to send files individually large uploads would fail less often if they were doing them per-file instead of all in one giant bundle.

This is just a suggestion and open to discussion. What does @selimnairb think?